lib/elixir/test/elixir/kernel/warning_test.exs

```diff
@@ -1413,6 +1413,22 @@ defmodule Kernel.WarningTest do
     purge(Sample)
   end
 
+  test "registered attribute with no use" do
```

Comment:
I put this here, because the test that warns about this issue is here. But, this test is checking that there is no warn, so it feels a bit weird.

Reply:
To assert there is no warning, you just write a regular test in module_test.exs. The fact the suite won't emit a warning while running it is enough.

Reply:
In any case, I think we should warn in the case below and I don't understand why we don't. :) The case we should not warn is if the attribute is registered and nothing is done with it.